spinn_utilities/make_tools/file_convertor.py

```diff
@@ -0,0 +1,453 @@
+            return True
+        if match.end(0) < len(stripped):
+            # Stuff after the log_end so check by char
+            return False
+
+        self._log_lines += 1
+        self._log_full += stripped
+        self._write_log_method(dest_f)
+        self._status = NORMAL_CODE
+        return True
+
+    def _process_line_in_log_close_bracket(self, dest_f):
+        stripped = self._text.strip()
+        if stripped[0] == ";":
+            if stripped == ";":
+                self._log_full += (";")
+                self._log_lines += 1
+                self._write_log_method(dest_f)
+                self._status = NORMAL_CODE
+                return True
+            else:
+                return False
+
+        elif stripped.startswith("//"):
+            # Just a comment line so write and move on
+            dest_f.write(self._text)
+            return True
+
+        else:
+            # so not a closing bracket so set status back
+            self._status = IN_LOG
+            return self._process_line_in_log(dest_f)
+
+    def _process_line_normal_code(self, dest_f):
+        # Fast check
+        if "log_" not in self._text:
+            # No log start found
+            dest_f.write(self._text)
+            return True
+
+        # Full slower check
+        stripped = self._text.strip()
+        match = LOG_START_REGEX.search(stripped)
+        if not match:
+            # No log start found after all
+            dest_f.write(self._text)
+            return True
+
+        if match.start() > 0:
+            if stripped.startswith("//"):
+                # Just a comment line so write and move on
+                dest_f.write(self._text)
+                return True
+            # Stuff before the log_start so check by character
+            return False
+
+        if LOG_START_REGEX.search(stripped, 1):
+            # Second start found so check by character
+            return False
+
+        # remove whitespaces and save log command
+        self._log_start = self._text.index(match.group(0))
+        self._log = "".join(match.group(0).split())
+        self._status = IN_LOG
+        self._log_full = ""  # text saved in process_line_in_log
+        self._log_lines = 0
+        # Now check for the end of log command
+        return self._process_line_in_log(dest_f)
+
+    def shorten(self, text):
+        count = text.count("%")
+        if count == 0:
+            return '"%u", {}'.format(self._message_id)
+        else:
+            result = '"%u'
+            matches = FORMATEXP.findall(text)
+            if len(matches) != count:
+                raise Exception("Unexpected formatString in {}".format(text))
+            for match in matches:
+                result += TOKEN
+                result += match
+            return result + '", {}'.format(self._message_id)
+
+    def _write_log_method(self, dest_f, tail=""):
+        self._message_id += 1
+        self._log_full = self._log_full.replace('""', '')
```

Comment:
missing in constructor self._log_full

Reply:
missing self._log_lines in constructor

Reply:
Added to slots but rather not add to init